File: app/src/main/java/com/jcr/sharedtasks/repository/SignInRepository.java
package com.jcr.sharedtasks.repository;

import android.content.SharedPreferences;

import com.google.firebase.auth.FirebaseUser;
import com.jcr.sharedtasks.AppExecutors;
import com.jcr.sharedtasks.db.ProjectsDao;
import com.jcr.sharedtasks.db.SharedTasksDb;

import javax.inject.Inject;

public class SignInRepository {

    private final SharedTasksDb db;

    private final ProjectsDao projectsDao;

    private final AppExecutors appExecutors;

    private final SharedPreferences sharedPreferences;

    @Inject
    public SignInRepository(AppExecutors appExecutors, SharedTasksDb db, ProjectsDao projectsDao,
                            SharedPreferences sharedPreferences) {
        this.db = db;
        this.projectsDao = projectsDao;
        this.appExecutors = appExecutors;
        this.sharedPreferences = sharedPreferences;
    }

    public void onSignedInitialize(FirebaseUser user) {
        sharedPreferences.edit().putString("userUid", user.getUid()).apply();
    }
}

File: app/src/main/java/com/jcr/sharedtasks/repository/SignInRepository.kt
package com.jcr.sharedtasks.repository

import android.content.SharedPreferences
import com.google.firebase.auth.FirebaseUser
import com.jcr.sharedtasks.testing.OpenForTesting
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
@OpenForTesting
class SignInRepository @Inject
constructor(private val sharedPreferences: SharedPreferences) {

    fun onSignedInitialize(user: FirebaseUser) {
        sharedPreferences.edit().putString("userUid", user.uid).apply()
        sharedPreferences.edit().putString("userName", user.displayName).apply()
    }
}
